raft/raft.hh

```diff
@@ -450,12 +442,44 @@ struct entry_id {
     index_t idx;
 };
 
+// The execute_add_entry/execute_modify_config methods can return this error to signal
+// that the request should be retried.
+// The exception is only used internally for entry/config forwarding and should not be leaked to a user.
+struct transient_error: public error {
+    // for IDL serialization
+    sstring message() const {
+        return what();
+    }
+    // Should the client delay before retrying.
+    bool delay;
```

Comment:
I don't understand the reasoning behind this `bool`, when does it make sense not to wait? It should be explained somewhere, in a comment and in the commit message.